extensions/cloud/portability-cloud-google/src/main/java/org/datatransferproject/cloud/google/GoogleJobStore.java

```diff
@@ -327,10 +300,64 @@ private Key getDataKey(UUID jobId, String key) {
     if (entity == null) {
       return null;
     }
+    String serializedEntity = entity.getString(type.getName());
     try {
-      return objectMapper.readValue(entity.getString(type.getName()), type);
+      return objectMapper.readValue(serializedEntity, type);
     } catch (IOException t) {
-      throw new RuntimeException("Failed to deserialized entityKey: " + entityKey, t);
+      throw new RuntimeException("Failed to deserialize entity: " + serializedEntity, t);
     }
   }
+
+  @Override
+  public void create(UUID jobId, String key, InputStream stream) {
+    String keyName = getDataKeyName(jobId, key);
+    // TODO: use result to determine success/failure
+    googleUserDataStore.create(keyName, stream);
+  }
+
+  @Override
+  public InputStream getStream(UUID jobId, String key) {
+    String keyName = getDataKeyName(jobId, key);
+    com.google.cloud.storage.Blob blob = googleUserDataStore.get(keyName);
```

Comment:
there is still more blob logic here... can the UserDataStore just take in jobId and key and figure out the rest itself? including getDataKeyName, that seems like logic particular to UserDataStore